chips/nrf52/src/nfct.rs

```diff
@@ -0,0 +1,703 @@
+            self.handle_txend();
+        }
+        if events_to_process.is_set(Interrupt::SELECTED) {
+            self.handle_selected();
+        }
+        if events_to_process.is_set(Interrupt::TXFRAMESTART) {
+            self.handle_txstart();
+            // Enable TXFRAMEEND
+            self.registers.intenset.modify(Interrupt::TXFRAMEEND::SET);
+        }
+        self.clear_errors();
+    }
+
+    fn active_events(&self) -> InMemoryRegister<u32, Interrupt::Register> {
+        let result = InMemoryRegister::new(0);
+        if NfcTag::take_event(&self.registers.event_ready) {
+            result.modify(Interrupt::READY::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_fielddetected) {
+            result.modify(Interrupt::FIELDDETECTED::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_fieldlost) {
+            result.modify(Interrupt::FIELDLOST::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_txframestart) {
+            result.modify(Interrupt::TXFRAMESTART::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_txframeend) {
+            result.modify(Interrupt::TXFRAMEEND::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_rxframestart) {
+            result.modify(Interrupt::RXFRAMESTART::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_rxframeend) {
+            result.modify(Interrupt::RXFRAMEEND::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_error) {
+            result.modify(Interrupt::ERROR::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_rxerror) {
+            result.modify(Interrupt::RXERROR::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_endrx) {
+            result.modify(Interrupt::ENDRX::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_endtx) {
+            result.modify(Interrupt::ENDTX::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_autocolresstarted) {
+            result.modify(Interrupt::AUTOCOLRESSTARTED::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_collision) {
+            result.modify(Interrupt::COLLISION::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_selected) {
+            result.modify(Interrupt::SELECTED::SET);
+        }
+        if NfcTag::take_event(&self.registers.event_started) {
+            result.modify(Interrupt::STARTED::SET);
+        }
+        result
+    }
+
+    // Reads the status of an Event register and clears the register.
+    // Returns the READY status.
+    fn take_event(event: &ReadWrite<u32, Event::Register>) -> bool {
+        let result = event.is_set(Event::GENERATED);
+        if result {
+            event.write(Event::GENERATED::CLEAR);
+        }
+        result
+    }
+
+    fn disable_all_interrupts(&self) {
+        self.registers.intenclr.set(0xffffffff);
```

Comment:
Does this disable interrupts or just clear them?

Reply:
It disables them.
Nordic has 3 registers for interrupt handling: `INTEN`, `INTENSET` and `INTENCLR`. The last two are only SR flip-flops modifying the `INTEN` register. It's the same as writing `self.registers.inten.set(0)` if you think this is clearer for the reviewer.

Reply:
`self.registers.inten.set(0)` seems clearer to me, but if what's already there actually does disable them that's fine.